walkeeper/src/replication.rs

```diff
@@ -121,7 +122,12 @@ impl ReplicationConn {
                     }
                 }
                 FeMessage::Sync => {}
-                FeMessage::CopyFail => return Err(anyhow!("Copy failed")),
+                FeMessage::CopyFail => {
+                    // Shutdown the connection, because rust-postgres client cannot be dropped
+                    // when connection is alive.
+                    let _ = stream_in.shutdown(Shutdown::Both);
```

Comment:
Could you please add a comment describing why it is ok to ignore shutdown error? I see that we return error anyway, but maybe it makes sense to log shutdown error too

Reply:
https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.shutdown

I think it may only return error if socket was already shut down somewhere else, which is ok for this case.